src/self-update/README.md

```diff
@@ -0,0 +1,89 @@
+  * When the plugin manages an update request for a package with module type "apt" it removes the package from the blacklist.
+  * When the plugin is called with LIST command for module type "apt", all non tedge packages (all installed but not blacklisted ones) will be reported.
+  * When the plugin is called with LIST command for module type "tedge", all tedge packages (all installed and blacklisted ones) will be reported.
+
+
+### Details about the SM Agent
+
+* The SM agent must accept the final exit code of a plugin execution via MQTT retain message on topic `tedge/plugins/software/<plugin name>`, send according update result message to the mapper, and clear the retain message on the topic afterwards.
+
+* For module type "tedge" the SM Agent must consider that a restart of the agent is valid.
+  * When the SM Agent flags an upcoming update request in the `persistance_store` before the request starts, it must store also the module type.
+  * When the SM Agent starts and finds a flagged update request in the `persistance_store` that is just for module type "tedge", no failure must be reported to the cloud.<br/>
+    TODO: An update request that contains module type "tedge" and others shall be rejected by the SM Agent with an error message. See Options below.
+
+# Options for more Robustness
+NOTE: Options below to be decided and addressed in a 2nd drop.
+
+### Robustness in Packages' installation logic
+* Use of external tools (e.g. grep, sed, ...) shall be avoided in package's installation logic. Instead Rust code/libraries shall be used.
+  * That is to reduce external dependencies and so reduce risc for failures.
+* At very beginning of the installation/removal logic all required/touched external ressources (e.g. config files as "mosquitto.conf", external tools (if any left), ...) shall be checked for availibility/accessibility.
+* If one external ressource is not available installation/removal procedure shall fail before any action was performed.
+
+### Robustness in Self-Update processing
+* When "tedge" plugin was started, no other update request for any other plugin must be in progress or started by the SM Agent.
+  * Reason: APT will restart the SM Agent during upgrade, what would result into half-executed update procedures of other parallel running plugins and could lead to corrupted system behaviour.
+  * It would be valid to execute any update requests for other plugins when tedge has completely finished. 
+    It would be also valid to reject any other update request and report an error to Cloud while tedge update is running. 
+* When executing APT the plugin will disable auto dependency solving.
+  * This is to avoid installing more than the requested packages to reduce risc for potential failures.
+* SM Agent shall fail when plugin for "tedge" does not support "update_list".
+  * This is since an update request will fail when there is more than one package and the agent is concerned
```

Comment:
After seeing the complexities in combining tedge update along with other package updates, I'm also in favour of not doing that in the initial drop. The UX of doing everything together is definitely better, but we could do that enhancement a later drop if acceptable. I honestly don't feel that isolated tedge self update is a deal breaker.

Reply:
There is one significant reason to use command "update_list". That is since the SM agen will restart during the update procedure (as it is part of the update request).

Lets imaging the update request as below:

    Package to update: tedge_apt_plugin.deb, tedge_agent.deb, tedge_mapper.deb
    
When using the package-wise commands "install" and "remove" instead of "update_list", the flow would look as below:
   1) Agent calls: `apt-plugin install tedge_apt_plugin.deb`
   2) Agent calls: `apt-plugin install tedge_agent.deb`
   3) Agent restarts due to installation of tedge_agent.deb
   4) Package tedge_mapper.deb would not be processed since Agent had restared.